internal/connector/connector.go

```diff
@@ -211,6 +215,9 @@ func Run(ctx context.Context, options Options) error {
 		TLSConfig:         tlsConfig,
 		Handler:           router,
 		ErrorLog:          httpErrorLog,
+		BaseContext: func(l net.Listener) context.Context {
+			return ctx
+		},
```

Comment:
so the http server exits when the main context is cancelled.

Reply:
Ah, but then we still have the issue of many 401 server requests. Right. I think even if we shutdown the connector we still have that problem because the pod will be restarted and the requests will start again.

Maybe we should use exponential backoff when there are failures? Or we could stop making requests and keep logging an error that says to fix the access key?

Reply:
Nice catch on the shutdown. I'll improve the code around server shutdown.
At least if the pod is failing it's likely to trigger some attention from the admin.

Reply:
Thinking about this more, is the intent here to block requests to the connector so that a connector that is not in contact with the infra server can't continue to proxy requests to kube? I think that's an important thing to do, but I wonder:
1. should allow some grace period between the first failure and when we start refusing connections?
2. should we also do this for other errors, not just 401. If the connector has failed to query the API for any reason, it should have the same behaviour after some period of time
3. should we keep the http server running, but return a clear error message like "infra connector has failed to communicate with the infra server for x seconds. Please contact your administrator to fix the problem."

Reply:
I created #3079 to track further improvements